### DISCUSSION TITLE
What is the intended way to represent views as assets created in dbt?

### DISCUSSION THREAD

**Discussion by mmutso-boku** [dagster-io/dagster#16388] [Q&A] [answered]
The scenario is that I have some source tables that are updated hourly, I have views created in dbt that use these source tables, and finally I have a daily running query that uses the views. Visualized it would look like this:

![image](https://github.com/dagster-io/dagster/assets/116641288/3c3d8e93-3d71-4ffa-b756-8d64a5b28ac3)

Logically, the view models/assets would be unpartitioned, because well, view is created once and that is it, there is nothing to partition. This however creates a problem - if for example in the source_table assets, a 2 days old partition is rematerialized, then the downstream view asset rematerialized, as expected because of the AllPartitionsMapping, but then for the downstream "model", the lastest partition is rematerialized (because of the LastPartitionMapping) and the 2 days old partition remains logically stale.

The workaround for this is making the view asset also hourly, but this is a bit ugly, and I was wondering how have other people solved this scenario?

**Comment by mmutso-boku** [comment]
Surprised that no-one else has encountered this?

**Comment by sryza** [comment, marked as answer]
Here's a relevant Github issue: https://github.com/dagster-io/dagster/issues/6474

**Comment by HuyNguyenNeyuData** [comment]
I got this same issues. I don't want to rebuild view since in Postgres, dbt will drop and re-create view => drop all dependent views
